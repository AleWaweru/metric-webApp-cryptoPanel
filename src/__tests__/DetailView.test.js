import React from 'react';
import { render } from '@testing-library/react';
import DetailsView from '../Components/Main/DetailView';

describe('DetailsView', () => {
  const props = {
    marketCapRank: 1,
    currentPrice: 40000,
    marketCap: 800000000,
    marketCapChange24h: 20000000,
    high24h: 45000,
    low24h: 35000,
    priceChange24h: 5000,
    priceChangePercentage24h: 10,
    marketCapChangePercentage24: 2,
    ath: 50000,
    athChangePercentage: 25,
    atl: 30000,
    atlChangePercentage: -10,
    fullyDilutedValuation: 1000000000,
    totalVolume: 100000,
    totalSupply: 50000000,
    circulatingSupply: 40000000,
    maxSupply: 60000000,
    publicInterestScore: 8,
  };

  it('renders the component correctly', () => {
    const { container } = render(
      <DetailsView
        marketCapRank={props.marketCapRank}
        currentPrice={props.currentPrice}
        marketCap={props.marketCap}
        marketCapChange24h={props.marketCapChange24h}
        high24h={props.high24h}
        low24h={props.low24h}
        priceChange24h={props.priceChange24h}
        priceChangePercentage24h={props.priceChangePercentage24h}
        marketCapChangePercentage24={props.marketCapChangePercentage24}
        ath={props.ath}
        athChangePercentage={props.athChangePercentage}
        atl={props.atl}
        atlChangePercentage={props.atlChangePercentage}
        fullyDilutedValuation={props.fullyDilutedValuation}
        totalVolume={props.totalVolume}
        totalSupply={props.totalSupply}
        circulatingSupply={props.circulatingSupply}
        maxSupply={props.maxSupply}
        publicInterestScore={props.publicInterestScore}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
