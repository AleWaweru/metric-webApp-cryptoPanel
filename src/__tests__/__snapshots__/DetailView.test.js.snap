// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsView renders the component correctly 1`] = `
<div>
  <div
    class="details-view"
  >
     
    <ul>
      <li>
        <span
          class="label"
        >
           
          Rank
          :
        </span>
        <span
          class="value"
        >
           
          1
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Price
          :
        </span>
        <span
          class="value"
        >
           
          $ 40K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Market Cap
          :
        </span>
        <span
          class="value"
        >
           
          $ 800M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Market Cap Change
          :
        </span>
        <span
          class="value"
        >
           
          $ 20M / 2 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          24H High Price
          :
        </span>
        <span
          class="value"
        >
           
          $ 45K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          24H Low Price
          :
        </span>
        <span
          class="value"
        >
           
          $ 35K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          24H Price Change
          :
        </span>
        <span
          class="value"
        >
           
          $ 5K / 10 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          All Time High Value
          :
        </span>
        <span
          class="value"
        >
           
          $ 50K / 25 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          All Time Low Value
          :
        </span>
        <span
          class="value"
        >
           
          $ 30K / -10 %
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Delutated Valuation
          :
        </span>
        <span
          class="value"
        >
           
          $ 1B
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Total Volume
          :
        </span>
        <span
          class="value"
        >
           
          $ 100K
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Total Supply
          :
        </span>
        <span
          class="value"
        >
           
          $ 50M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Circulating Supply
          :
        </span>
        <span
          class="value"
        >
           
          $ 40M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Maximum Supply
          :
        </span>
        <span
          class="value"
        >
           
          $ 60M
        </span>
      </li>
      <li>
        <span
          class="label"
        >
           
          Interest Score
          :
        </span>
        <span
          class="value"
        >
           
          8
        </span>
      </li>
    </ul>
  </div>
</div>
`;
